Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/viewholder/CheckBoxSettingViewHolder.java

```diff
@@ -36,7 +38,15 @@ public void bind(SettingsItem item)
   {
     mItem = (CheckBoxSetting) item;
 
-    mTextSettingName.setText(item.getNameId());
+    // Special case for LogTypes retrieved via JNI since those aren't string references.
+    if (item.getNameId() == 0 && item.getSection().equals(Settings.SECTION_LOGGER_LOGS))
+    {
+      mTextSettingName.setText(NativeLibrary.GetLogTypeNames().get(item.getKey()));
```

Comment:
This is a rather inefficient use of a map. You're forcing the whole map to get reconstructed each time you access one value of it.